### PR TITLE
Introduce target_stack property to update specific stacks and enable finder preview

### PR DESCRIPTION
**Pull request by minhngocd**
Introduced `target_stack` property to replace pre_production flag
    
The new `target_stack` property will publish anything with a target stack of `live` to the live stack and draft stack and anything else to the draft stack only.
This allows us to surface appropriate preview links and live links for users later.

Delete anything associated with pre_production and updated documentation

https://trello.com/c/JrkAwfDU/2786-dynamically-update-stacks-based-on-state-of-specialist-finder    

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: lib/documents/schemas/aaib_reports.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+  "target_stack": "live",
```
Comment (ChrisBAshton) [inline lib/documents/schemas/aaib_reports.json:2]: To make it easier to review, I'd move the config update into its own commit (i.e. add all the `target_stack` properties in one commit, then the logic that uses it would be added in the next commit).

File: lib/documents/schemas/veterans_support_organisations.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "pre-production": true,
```
Comment (ChrisBAshton) [inline lib/documents/schemas/veterans_support_organisations.json]: I'd also remove the pre-production config and logic in a separate commit, once the `target_stack` stuff has been implemented.

EDIT: looks like you've already tried to do this - in which case, just double-check this and rebase it into the (currently) third commit.

File: docs/creating-editing-and-removing-specialist-document-types-and-finders.md [hunk 3]
```diff
@@ -72,17 +72,24 @@ If your email sign up page should have checkboxes (e.g. [cma-cases](https://www.
 
 ## 5. Deploy and publish <a id="deploy-and-publish"></a>
```
Comment (ChrisBAshton) [inline docs/creating-editing-and-removing-specialist-document-types-and-finders.md]: Is it worth splitting this out into:

```
## 5. Deploy a finder to the draft stack (for previewing)

## 6. Publish a finder
```

**Review by ChrisBAshton — COMMENTED** [review]
LGTM 👍 a few comments below.

**Comment by minhngocd** [comment]
I think i've addressed all the comments and rebased off of main @ChrisBAshton - it's ready for another review please 🙏

File: lib/documents/schemas/veterans_support_organisations.json [hunk 4]
```diff
@@ -1,4 +1,5 @@
 {
+  "target_stack": "draft",
```
Comment (ChrisBAshton) [inline lib/documents/schemas/veterans_support_organisations.json]: This should probably be `live`, unless there are any config changes here that aren't on the [live site](https://www.gov.uk/support-for-veterans) yet.

File: lib/tasks/publishing_api.rake [hunk 5]
```diff
@@ -12,19 +12,6 @@ namespace :publishing_api do
     end
   end
 
-  desc "Publish pre-production finders only, to the Publishing API"
```
Comment (ChrisBAshton) [inline lib/tasks/publishing_api.rake:15]: This removal should probably be fixed up to the "Remove usages of pre_production flag" commit.

File: spec/lib/publishing_api_finder_publisher_spec.rb [hunk 6]
```diff
@@ -109,12 +106,8 @@ def make_finder(base_path, overrides = {})
       end
     end
 
-    context "when the finder isn't `pre_production`" do
```
Comment (ChrisBAshton) [inline spec/lib/publishing_api_finder_publisher_spec.rb:112]: Ideally this (and the removals below) should also be in the "Remove usages of pre_production flag" commit, but I appreciate it may be hard to unpick as at this point we've changed the deployment logic 🤔

File: docs/creating-editing-and-removing-specialist-document-types-and-finders.md [hunk 7]
```diff
@@ -36,7 +36,7 @@ When the PR is reviewed and its tests passing, it can be merged and deployed at
 
 See [CMA cases](https://github.com/alphagov/specialist-publisher/blob/main/lib/documents/schemas/cma_cases.json).
 
-New formats are often requested to be deployed in "pre-production mode", which is configured in this step ([example](https://github.com/alphagov/specialist-publisher/blob/f8e93142dfad6f3971a73c923b01f2e7352bdb54/lib/documents/schemas/tax_tribunal_decisions.json#L64)). `pre-production` documents are only publishable on development and integration.
+New formats should be added with `state: "draft"` so that departments can preview the changes.
```
Comment (ChrisBAshton) [inline docs/creating-editing-and-removing-specialist-document-types-and-finders.md]:
```suggestion
New formats should be added with `state: "draft"` so that departments can preview the finder before you publish it.
```

File: docs/creating-editing-and-removing-specialist-document-types-and-finders.md [hunk 8]
```diff
@@ -70,20 +70,29 @@ The email sign up page is rendered by [Finder Frontend](https://github.com/alpha
...
-To deploy:
+## 5. Deploy a finder to the draft stack (for previewing)
 
+To deploy a new finder for previewing:
+   1. Ensure the finder target_stack is set to `draft`
    1. Deploy Publishing API (if you haven't already), Specialist Publisher and Search API. 
       - Ensure you deploy Publishing API first, to avoid schema validation errors.
       - Also deploy Email Alert API if you have made changes to it.
-   2. [Reindex the govuk Elasticsearch index](https://docs.publishing.service.gov.uk/manual/reindex-elasticsearch.html#how-to-reindex-an-elasticsearch-index).
+   1. [Reindex the govuk Elasticsearch index](https://docs.publishing.service.gov.uk/manual/reindex-elasticsearch.html#how-to-reindex-an-elasticsearch-index).
```
Comment (ChrisBAshton) [inline docs/creating-editing-and-removing-specialist-document-types-and-finders.md]: Hmmm - given it's only the finder 'skeleton' that's previewable, I'm not sure there's much point to the indexing here. Should this step (and its substeps) be moved into the publish section instead?

**Review by ChrisBAshton — COMMENTED** [review]
Thanks @minhngocd - looking good, just a few follow-up comments below.

File: lib/tasks/publishing_api.rake (hunk 5 above)
Comment (minhngocd) [reply lib/tasks/publishing_api.rake:15]: But the logic is switched and changed in this commit, so this rake task would be broken anyways - we thought it'd be best to remove in this commit

File: spec/lib/publishing_api_finder_publisher_spec.rb (hunk 6 above)
Comment (minhngocd) [reply spec/lib/publishing_api_finder_publisher_spec.rb:112]: Similarly here - if we leave these tests here they would be failing

File: lib/documents/schemas/veterans_support_organisations.json (hunk 4 above)
Comment (minhngocd) [reply lib/documents/schemas/veterans_support_organisations.json]: oooh thanks for catching this! i lost it amongst the merge conflicts

**Comment by minhngocd** [comment]
@ChrisBAshton updated !! 🙏 for the test and rake task removal related to the pre-prod flag, I think it's better to go with the logic change commit otherwise we are left with some broken tests or logic in the interim.

**Review by ChrisBAshton — APPROVED** [review]
Nice job 👏
